core/templates/pages/signup-page/signup-page.component.ts

```diff
@@ -159,7 +159,7 @@ export class SignupPageComponent {
   submitPrerequisitesForm(
       agreedToTerms: boolean,
       username: string,
-      canReceiveEmailUpdates: string | null): void {
+      canReceiveEmailUpdates: string | null = null): void {
```

Comment:
can you please explain this?

Reply:
Hello! @heyimShivam Since when the user does not selects any preference for email canReceiveEmailUpdates becomes undefined and the code throws an exception 


```typescript
if (this.showEmailPreferencesForm && !this.hasUsername) {
      if (canReceiveEmailUpdates === null) {
        this.emailPreferencesWarningText = 'I18N_SIGNUP_FIELD_REQUIRED';
        return;
      }

      if (canReceiveEmailUpdates === 'yes') {
        requestParams.can_receive_email_updates = true;
      } else if (canReceiveEmailUpdates === 'no') {
        requestParams.can_receive_email_updates = false;
      } else {
        throw new Error(
          'Invalid value for email preferences: ' +
          canReceiveEmailUpdates);
      }
    }
        
```


as you can see above if the value is undefined we get an error else the situation is handled and the user is displayed a warning to select a preference for email.

So that's why i have provided a default value as null.